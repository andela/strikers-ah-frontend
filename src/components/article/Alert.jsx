/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export const Alert = ({ alert }) =>
  alert !== null &&
  alert.length > 0 &&
  alert.map(alt => (
    <div key={alt.id} className={`alert alert-${alt.alertType}`}>
      {alt.msg}
    </div>
  ));
Alert.propTypes = {
  alert: PropTypes.array.isRequired,
};
const mapStateToProps = state => ({
  alert: state.alertReducer,
});

export default connect(mapStateToProps)(Alert);
