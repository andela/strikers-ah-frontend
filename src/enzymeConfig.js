import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
