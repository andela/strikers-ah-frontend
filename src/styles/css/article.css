@import url("https://fonts.googleapis.com/css?family=Raleway&display=swap");
.spiner {
  position: fixed;
  z-index: 1031;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: #43c0f1; }

.comments-container {
  padding: 1%;
  margin-top: 1%;
  border-radius: 1%; }
  .comments-container .comment-count {
    font-weight: bold;
    margin-left: 4%; }

html {
  height: 100%;
  margin: 0; }

body {
  font-family: "Arial", "Helvetica", "sans-serif";
  font-size: 15px;
  margin: 0;
  padding: 0;
  /* background-color: darkgray; */
  min-height: 100%; }

.container {
  width: 80%;
  margin: auto;
  overflow: hidden; }
  @media (max-width: 768px) {
    .container {
      width: 100%; } }

header {
  background-color: #ffffff;
  padding-top: 10px;
  min-height: 70px;
  border-bottom: #e5e5e5 3px solid; }

.brand {
  margin-top: 10px;
  float: left; }

header .icons ul {
  list-style-type: none;
  margin: 0;
  padding: 0; }

header .icons ul li i {
  color: grey; }

header .icons {
  margin-top: 10px;
  display: flex;
  flex-direction: row;
  justify-content: flex-end; }
  @media (max-width: 768px) {
    header .icons {
      display: none; } }

.brand .brandName {
  text-decoration: none; }

.brand .brandName {
  font-size: 25px;
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: #00aef9; }

.content {
  background-color: #e5e5e5;
  align-content: center;
  margin-right: 100px;
  margin-left: 100px; }
  @media (max-width: 768px) {
    .content {
      margin: auto;
      margin-right: 5%;
      margin-left: 5%; } }

.content .editor-tools .container .bar-tool #align-bartool {
  text-align: center; }

.content .editor-tools .container .bar-tool {
  margin-top: 10px;
  margin-bottom: 10px;
  background-color: #ffffff;
  border-radius: 8px;
  box-shadow: 0px 4px 4px #c4c4c4;
  padding: 10px;
  display: inline-block;
  /* position: absolute; */ }
  @media (max-width: 768px) {
    .content .editor-tools .container .bar-tool {
      margin: auto;
      padding: 5px; } }

.content .editor-tools .container .bar-tool {
  margin-bottom: 100px;
  margin-left: 10%;
  margin-right: 10%; }

input[type='color'] {
  border: none; }

.content .editor-tools .container .bar-tool #border {
  border: 1px solid #c4c4c4;
  transform: rotate(90deg); }

.content .editor-tools .container .bar-tool > button {
  margin-right: 5px;
  margin-left: 5px;
  border: no; }

.content .editor-tools .container .bar-tool > button:hover {
  background-color: #dfe6e9;
  transition: all 0.3s linear 0s;
  cursor: pointer; }

.content .editor-tools .container .bar-tool #highlight {
  cursor: pointer; }

.content .editor-tools .container .bar-tool #highlight {
  margin-right: 5px;
  margin-left: 5px; }

.content .text-area {
  width: 100%; }

[contentEditable='true']:empty:not(:focus):before {
  content: attr(data-text);
  color: gray; }

.content .text-area .form #title {
  background-color: #ffffff;
  margin-bottom: 5px;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: "Arial", "Helvetica", "sans-serif";
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%; }

.content .text-area .form #body {
  margin-bottom: 10px;
  background-color: #ffffff;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: "Arial", "Helvetica", "sans-serif";
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%;
  min-height: 800px; }

.content .text-area .form #publish {
  cursor: pointer;
  border: none;
  background: #808e9b;
  font-size: 16px;
  padding: 15px 32px;
  display: inline-block;
  text-align: center;
  color: #ffffff;
  border-radius: 5px;
  float: right;
  margin-bottom: 20px; }

nav.icons div {
  margin-right: 15px; }

nav.icons i {
  margin-top: 5px; }

nav.icons div.search input[type='text'] {
  margin-right: 10px;
  border-radius: 6px;
  padding: 4px 12px;
  border-color: #eee;
  border-width: 1px; }

nav.icons div.search #search-button {
  margin-right: 10px; }

.card {
  float: left;
  margin-top: 10px;
  padding: 10px;
  width: 25%;
  height: 400px;
  background-color: #ffffff;
  border-color: #00aef9;
  border-width: 1px;
  box-shadow: #808e9b;
  margin-right: 10px; }

.card:hover {
  cursor: pointer;
  text-decoration: none; }

.card > img {
  width: 100%;
  height: 50%; }

.card .contentDescription {
  width: 100%;
  margin-top: 10px;
  word-wrap: break-word; }

.card .contentDescription .articleCardHeader {
  width: 100%;
  text-align: center;
  word-wrap: break-word; }

.card .contentDescription .articleCardHeader p {
  font-weight: bold;
  font-size: 20px;
  text-transform: capitalize; }

div.alert {
  width: 30%;
  height: 30px;
  display: inline-block;
  background-color: #bb321f;
  color: #ffffff;
  margin-left: 30%;
  margin-right: 30%;
  border-radius: 5px;
  margin-top: 10px;
  padding: 30px;
  font-size: 20px; }

.contents {
  background-color: #ffffff;
  margin-top: 0;
  min-height: 800px; }

.contents .contentTitle {
  width: 100%;
  text-align: center;
  padding: 10px; }

.contents .contentBody #articleBody {
  clear: both;
  float: left;
  margin-top: 15px;
  padding-left: 10%;
  padding-right: 10%; }

.contents .contentBody p {
  padding-left: 50px;
  padding-right: 50px;
  font-size: 20px;
  word-wrap: break-word;
  line-height: 1.5em; }

.contents .contentBody p img {
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  display: block;
  height: 500px; }

.article-icons .icon {
  margin-right: 10px;
  display: inline-block; }

.contentDescription a {
  text-decoration: none; }

.contentDescription a:link {
  color: #2f3640; }

.contentDescription a:visited {
  color: #2f3640; }

.card .contentDescription .buttons {
  margin-bottom: 5px;
  position: absolute;
  margin-top: 10px; }

.card .contentDescription .buttons .redButton {
  background: #bb321f;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px; }

.card .contentDescription .buttons .lightBlueButton {
  background: #74b9ff;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px; }

.card .articleDescription {
  height: 65px;
  margin-top: 5px; }

.container .allArticlecontents {
  background-color: #ffffff;
  margin-top: 0;
  height: 100%; }

.socialMedia {
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer; }
  .socialMedia .facebookButton {
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
    .socialMedia .facebookButton.facebook {
      color: #3b5998;
      border-color: #3b5998; }
      .socialMedia .facebookButton.facebook:hover {
        background-color: #3b5998;
        border-color: #3b5998;
        color: #fff; }
    .socialMedia .facebookButton.twitter {
      color: #1da1f2;
      border-color: #1da1f2; }
      .socialMedia .facebookButton.twitter:hover {
        background-color: #1da1f2;
        border-color: #1da1f2;
        color: #fff; }
    .socialMedia .facebookButton.googleplus {
      color: #dd4b39;
      border-color: #dd4b39; }
      .socialMedia .facebookButton.googleplus:hover {
        background-color: #dd4b39;
        border-color: #dd4b39;
        color: #fff; }
    .socialMedia .facebookButton.linkedin {
      color: #0077b5;
      border-color: #0077b5; }
      .socialMedia .facebookButton.linkedin:hover {
        background-color: #0077b5;
        border-color: #0077b5;
        color: #fff; }
    .socialMedia .facebookButton.whatsapp {
      color: #128c7e;
      border-color: #128c7e; }
      .socialMedia .facebookButton.whatsapp:hover {
        background-color: #128c7e;
        border-color: #128c7e;
        color: #fff; }
  .socialMedia .twitterButton {
    color: #1da1f2;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .twitterButton:hover {
    background-color: #1da1f2;
    color: #fff; }
  .socialMedia .linkedinButton {
    color: #0077b5;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .linkedinButton:hover {
    background-color: #0077b5;
    color: #fff; }
  .socialMedia .whatsappButton {
    color: #128c7e;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .whatsappButton:hover {
    color: #fff;
    background-color: #128c7e; }
  .socialMedia .emailButton {
    color: #dd4b39;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .emailButton:hover {
    color: #fff;
    background-color: #dd4b39; }

.edit-history-section {
  margin-top: 10px; }
  .edit-history-section .hideButton {
    background: #fff;
    padding: 3px;
    color: #000;
    border-radius: 3px;
    cursor: pointer;
    margin-bottom: 3px;
    font-size: 20px; }

.history-section {
  padding: 3px 15px 5px 0px;
  border-bottom: 1px solid #c5c5c5; }
  .history-section time {
    color: #9c9c9c;
    font-size: 12px;
    font-weight: bold; }

.emailButton:hover {
  color: #fff;
  background-color: #dd4b39; }

.likeButton {
  border: none;
  color: #9c9c9c;
  cursor: pointer;
  width: 15px;
  height: 15px; }

.likeButton:hover {
  color: #43c0f1; }

.buttonSpan {
  margin-right: 5px; }

.topImage input[type='file'] {
  cursor: inherit;
  display: block;
  min-height: 100%;
  min-width: 100%;
  opacity: 0;
  position: absolute;
  right: 0;
  text-align: right;
  top: 0;
  float: right; }

.topImage .file-upload {
  overflow: hidden;
  position: absolute;
  width: 100px;
  height: 100px; }

.topImage .file-upload .imageIcone {
  min-height: 100%;
  min-width: 100%;
  margin-left: 20%;
  color: #3b5998; }

.tags {
  padding: 10px;
  margin-left: 20px;
  background-color: #3b5998;
  color: #fff;
  border-radius: 5px;
  display: inline-block;
  cursor: pointer; }

#tags > input {
  border-radius: 5px;
  border: none;
  width: 178px;
  height: 36px;
  margin-bottom: 5px;
  padding: 10px;
  font-size: 15px;
  margin-right: 5px; }

#tags {
  width: 75%;
  float: left; }

#addTag {
  width: 10%;
  float: left; }

.tagList {
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
  background-color: #4b4b4b;
  color: #fff;
  display: inline-block;
  padding: 5px;
  cursor: pointer; }
