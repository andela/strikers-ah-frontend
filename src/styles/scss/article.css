@import url('https://fonts.googleapis.com/css?family=Raleway&display=swap');
.comments-container {
  padding: 1%;
  margin-top: 1%;
  border-radius: 1%;
}

.comments-container .comment-count {
  font-weight: bold;
  margin-left: 4%;
}

html {
  height: 100%;
  margin: 0;
}

body {
  font-family: 'Arial', 'Helvetica', 'sans-serif';
  font-size: 15px;
  margin: 0;
  padding: 0;
  /* background-color: darkgray; */
  min-height: 100%;
}

.container {
  width: 80%;
  margin: auto;
  overflow: hidden;
}

@media (max-width: 768px) {
  .container {
    width: 100%;
  }
}

header {
  background-color: #ffffff;
  padding-top: 10px;
  min-height: 70px;
  border-bottom: #e5e5e5 3px solid;
}

.brand {
  margin-top: 10px;
  float: left;
}

header .icons ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

header .icons ul li i {
  color: grey;
}

header .icons {
  margin-top: 10px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

@media (max-width: 768px) {
  header .icons {
    display: none;
  }
}

.brand .brandName {
  text-decoration: none;
}

.brand .brandName {
  font-size: 25px;
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: #00aef9;
}

.content {
  background-color: #e5e5e5;
  -ms-flex-line-pack: center;
  align-content: center;
  margin-right: 100px;
  margin-left: 100px;
}

@media (max-width: 768px) {
  .content {
    margin: auto;
    margin-right: 5%;
    margin-left: 5%;
  }
}

.content .editor-tools .container .bar-tool #align-bartool {
  text-align: center;
}

.content .editor-tools .container .bar-tool {
  margin-top: 10px;
  margin-left: 30%;
  margin-right: 30%;
  margin-bottom: 10px;
  background-color: #ffffff;
  border-radius: 8px;
  -webkit-box-shadow: 0px 4px 4px #c4c4c4;
  box-shadow: 0px 4px 4px #c4c4c4;
  padding: 10px;
  display: inline-block;
  /* position: absolute; */
}

@media (max-width: 768px) {
  .content .editor-tools .container .bar-tool {
    margin: auto;
    padding: 5px;
  }
}

input[type='color'] {
  border: none;
}

.content .editor-tools .container .bar-tool #border {
  border: 1px solid #c4c4c4;
  -webkit-transform: rotate(90deg);
  transform: rotate(90deg);
}

.content .editor-tools .container .bar-tool > button {
  margin-right: 5px;
  margin-left: 5px;
  border: no;
}

.content .editor-tools .container .bar-tool > button:hover {
  background-color: #dfe6e9;
  -webkit-transition: all 0.3s linear 0s;
  transition: all 0.3s linear 0s;
  cursor: pointer;
}

.content .editor-tools .container .bar-tool #highlight {
  cursor: pointer;
}

.content .editor-tools .container .bar-tool #highlight {
  margin-right: 5px;
  margin-left: 5px;
}

.content .text-area {
  width: 100%;
}

[contentEditable='true']:empty:not(:focus):before {
  content: attr(data-text);
  color: gray;
}

.content .text-area .form #title {
  background-color: #ffffff;
  margin-bottom: 5px;
  -ms-flex-line-pack: center;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: 'Arial', 'Helvetica', 'sans-serif';
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%;
}

.content .text-area .form #body {
  margin-bottom: 10px;
  background-color: #ffffff;
  -ms-flex-line-pack: center;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: 'Arial', 'Helvetica', 'sans-serif';
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%;
  min-height: 800px;
}

.content .text-area .form #publish {
  border: none;
  background: #808e9b;
  font-size: 16px;
  padding: 15px 32px;
  display: inline-block;
  text-align: center;
  color: #ffffff;
  border-radius: 5px;
  float: right;
  margin-bottom: 20px;
}

nav.icons div {
  margin-right: 15px;
}

nav.icons i {
  margin-top: 5px;
}

nav.icons div.search input[type='text'] {
  margin-right: 10px;
  border-radius: 6px;
  padding: 4px 12px;
  border-color: #eee;
  border-width: 1px;
}

nav.icons div.search #search-button {
  margin-right: 10px;
}

.card {
  float: left;
  margin-top: 10px;
  padding: 10px;
  width: 25%;
  height: 400px;
  background-color: #ffffff;
  border-color: #00aef9;
  border-width: 1px;
  -webkit-box-shadow: #808e9b;
  box-shadow: #808e9b;
  margin-right: 10px;
}

.card:hover {
  cursor: pointer;
  text-decoration: none;
}

.card > img {
  width: 100%;
  height: 50%;
}

.card .contentDescription {
  width: 100%;
  margin-top: 10px;
  word-wrap: break-word;
}

.card .contentDescription .articleCardHeader {
  width: 100%;
  text-align: center;
  word-wrap: break-word;
}

.card .contentDescription .articleCardHeader p {
  font-weight: bold;
  font-size: 20px;
  text-transform: capitalize;
}

div.alert {
  width: 30%;
  height: 30px;
  display: inline-block;
  background-color: #bb321f;
  color: #ffffff;
  margin-left: 30%;
  margin-right: 30%;
  border-radius: 5px;
  margin-top: 10px;
  padding: 30px;
  font-size: 20px;
}

.contents {
  background-color: #ffffff;
  margin-top: 0;
  min-height: 800px;
}

.contents .contentTitle {
  width: 100%;
  text-align: center;
  padding: 10px;
}

.contents .contentBody p {
  padding-left: 50px;
  padding-right: 50px;
  font-size: 20px;
  word-wrap: break-word;
}

.contents .contentBody p img {
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  display: block;
  height: 500px;
}

.article-icons .icon {
  margin-right: 10px;
  display: inline-block;
}

.contentDescription a {
  text-decoration: none;
}

.contentDescription a:link {
  color: #2f3640;
}

.contentDescription a:visited {
  color: #2f3640;
}

.card .contentDescription .buttons {
  margin-bottom: 5px;
  position: absolute;
  margin-top: 10px;
}

.card .contentDescription .buttons .redButton {
  background: #bb321f;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px;
}

.card .contentDescription .buttons .lightBlueButton {
  background: #74b9ff;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px;
}

.card .articleDescription {
  height: 65px;
  margin-top: 5px;
}

.container .allArticlecontents {
  background-color: #ffffff;
  margin-top: 0;
  height: 100%;
}

.socialMedia {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.socialMedia .facebookButton {
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .facebookButton.facebook {
  color: #3b5998;
  border-color: #3b5998;
}

.socialMedia .facebookButton.facebook:hover {
  background-color: #3b5998;
  border-color: #3b5998;
  color: #fff;
}

.socialMedia .facebookButton.twitter {
  color: #1da1f2;
  border-color: #1da1f2;
}

.socialMedia .facebookButton.twitter:hover {
  background-color: #1da1f2;
  border-color: #1da1f2;
  color: #fff;
}

.socialMedia .facebookButton.googleplus {
  color: #dd4b39;
  border-color: #dd4b39;
}

.socialMedia .facebookButton.googleplus:hover {
  background-color: #dd4b39;
  border-color: #dd4b39;
  color: #fff;
}

.socialMedia .facebookButton.linkedin {
  color: #0077b5;
  border-color: #0077b5;
}

.socialMedia .facebookButton.linkedin:hover {
  background-color: #0077b5;
  border-color: #0077b5;
  color: #fff;
}

.socialMedia .facebookButton.whatsapp {
  color: #128c7e;
  border-color: #128c7e;
}

.socialMedia .facebookButton.whatsapp:hover {
  background-color: #128c7e;
  border-color: #128c7e;
  color: #fff;
}

.socialMedia .twitterButton {
  color: #1da1f2;
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .twitterButton:hover {
  background-color: #1da1f2;
  color: #fff;
}

.socialMedia .linkedinButton {
  color: #0077b5;
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .linkedinButton:hover {
  background-color: #0077b5;
  color: #fff;
}

.socialMedia .whatsappButton {
  color: #128c7e;
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .whatsappButton:hover {
  color: #fff;
  background-color: #128c7e;
}

.socialMedia .emailButton {
  color: #dd4b39;
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
}

.socialMedia .emailButton:hover {
  color: #fff;
  background-color: #dd4b39;
}

.edit-history-section {
  margin-top: 10px;
}

.edit-history-section .hideButton {
  background: #fff;
  padding: 3px;
  color: #000;
  border-radius: 3px;
  cursor: pointer;
  margin-bottom: 3px;
  font-size: 20px;
}

.history-section {
  padding: 3px 15px 5px 0px;
  border-bottom: 1px solid #c5c5c5;
}

.history-section time {
  color: #9c9c9c;
  font-size: 12px;
  font-weight: bold;
}

.emailButton:hover {
  color: #fff;
  background-color: #dd4b39;
}

.likeButton {
  border: none;
  color: #9c9c9c;
  cursor: pointer;
  width: 15px;
  height: 15px;
}

.likeButton:hover {
  color: #43c0f1;
}

.buttonSpan {
  margin-right: 5px;
}
/*# sourceMappingURL=article.css.map */
