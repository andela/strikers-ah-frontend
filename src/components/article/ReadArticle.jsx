/* eslint-disable import/no-named-as-default */
/* eslint-disable react/destructuring-assignment */
import reactHtml from 'html-react-parser';
import uuid from 'uuid';
import React, { Component, Fragment } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import Header from './Header';
import '../../styles/css/article.css';
import '../../styles/css/highlight.css';
// import '../../helpers/highlight';
import twitterIcon from '../../styles/img/twitter-icon.png';
import facebookIcon from '../../styles/img/facebook-icon.png';
import copyIcon from '../../styles/img/copy-icon.png';
import commentIcon from '../../styles/img/commenting-icon.png';
import { getOneArticle } from '../../redux/actions/articleAction';
import Author from './Author';
import SocialMedia from './SocialMedia';
import CommentForm from '../comment/commentForm';
import Spinner from './Spinner';
import {
  addComment,
  getComments,
  deleteComment,
  editComment,
} from '../../redux/actions/commentAction';
import Comment from '../comment/comment';
import Rate from '../common/Rate';
import Ratings from '../common/Ratings';

/**
 *@author: Innocent Nkunzi
 @param { Integer } commentId --
 @param { Boolaen } cancel
 * @returns {*} Articleform
 */
export class ReadArticle extends Component {
  state = {
    commentEditMode: false,
    editCommentId: 0,
    editHistoryMode: false,
    commentHistoryId: 0,
  };

  /**
   * @author Innocent Nkunzi
   * @returns {*} componentWillmount
   */
  componentWillMount() {
    const { slug } = this.props.match.params;
    this.props.getOneArticle(slug);
    this.props.getComments(slug);
  }

  /**
   * @author Mwibutsa Floribert
   * @param {*} nextProps
   * @returns { * } --
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.comment) {
      this.props.comments.unshift(nextProps.comment);
    }
  }

  toggleEditCommentForm = (commentId, cancel) => {
    this.setState({
      commentEditMode: true && !cancel,
      editCommentId: commentId,
    });
  };

  /**
   * @param {* } commentId --
   * @param { * } cancel --
   * @returns { * } --
   */
  toggleEditHistory = (commentId, cancel) => {
    this.setState({
      editHistoryMode: true && !cancel,
      commentHistoryId: commentId,
    });
  };

  /**
   * @author Innocent Nkunzi
   * @returns {*} component
   */
  render() {
    const id = uuid.v4();
    const {
      addComment: saveComment,
      deleteComment: removeComment,
      editComment: modifyComment,
    } = this.props;
    const {
      commentEditMode,
      editCommentId,
      editHistoryMode,
      commentHistoryId,
    } = this.state;
    const { slug } = this.props.match.params;
    const singleArticle = this.props.article.article;
    const {
      article: { message },
    } = this.props;
    if (
      message !== undefined &&
      message === 'No article found with the slug provided'
    ) {
      return <Redirect to="/not-found" />;
    }
    if (singleArticle !== null && singleArticle !== undefined) {
      let commentList = [];
      const { comments } = this.props;
      commentList = comments;
      import('../../helpers/highlight');
      return (
        <Fragment>
          <Header />
          <div className="comment-box-content-container">
            <p className="comment-box-content-container-text" />
            <fieldset>
              <button type="button" className="contact-cancel cancelCommentBox">
                Exit
              </button>
              <button
                type="button"
                className="contact-cancel"
                id="deletecommentBtn"
              >
                Delete
              </button>
            </fieldset>
          </div>
          <div className="comment-box-container">
            <form id="comment-box">
              <fieldset>
                <textarea
                  id="comment"
                  placeholder="Type your comment here...."
                  required
                />
              </fieldset>
              <fieldset>
                <button
                  type="button"
                  className="contact-cancel hideCommentBoxBtn"
                >
                  Cancel
                </button>
                <button type="button" className="contact-submit addCommentBtn">
                  Submit
                </button>
              </fieldset>
            </form>
          </div>

          <div className="highlight">
            <span className="section twitter">
              <img
                src={twitterIcon}
                alt="Twitter Icon"
                width="512.002px"
                height="512.002px"
              />
            </span>
            <span className="section facebook">
              <img
                src={facebookIcon}
                alt="Facebook Icon"
                width="512.002px"
                height="512.002px"
              />
            </span>
            <span className="section copyclipboard">
              <img
                src={copyIcon}
                alt="Copy Icon"
                width="512.002px"
                height="512.002px"
              />
            </span>
            <span className="section comment">
              <img
                src={commentIcon}
                alt="Comment Icon"
                width="512.002px"
                height="512.002px"
              />
            </span>
          </div>
          <div className="container">
            <div className="contents">
              <div className="contentTitle">
                <h1>{singleArticle.title && reactHtml(singleArticle.title)}</h1>
                <Author />
                <SocialMedia />
                {singleArticle.taglist !== undefined &&
                singleArticle.taglist.length !== 0
                  ? singleArticle.taglist.map(tag => (
                      <div className="tagList" key={id}>
                        #{tag}
                      </div>
                    ))
                  : ''}
              </div>
              <div className="contentBody">
                {singleArticle.image !== 'null' &&
                singleArticle.image !== undefined ? (
                  <p id="styleImage">
                    <img src={singleArticle.image} alt="Article" />
                  </p>
                ) : (
                  ''
                )}
                <p id="articleBody" className="articleBody">
                  {singleArticle.body && reactHtml(singleArticle.body)}
                </p>
                <hr />
                <div className="rating">
                  <Rate slug={singleArticle.slug} />
                  <Ratings slug={singleArticle.slug} />
                </div>
              </div>
            </div>
            <CommentForm
              slug={slug}
              saveComment={saveComment}
              buttonLabel="Comment"
            />
            <div className="comments-container">
              {comments && (
                <span className="comment-count">
                  {commentList.length > 0 && (
                    <span test-data="count-comment">
                      {commentList.length > 1
                        ? `${commentList.length} Comments`
                        : `${commentList.length} Comment`}
                    </span>
                  )}
                </span>
              )}
              {comments && (
                <span>
                  {commentList.length === 0 && (
                    <span>Be the first to add a comment</span>
                  )}
                  <br />
                </span>
              )}
              {comments
                ? commentList.length > 0 &&
                  commentList.map(comment => {
                    return (
                      <Comment
                        formId={editCommentId}
                        comment={comment}
                        key={comment.id}
                        editComment={modifyComment}
                        deleteComment={removeComment}
                        slug={slug}
                        toggleEditCommentForm={(cancel = false) =>
                          this.toggleEditCommentForm(comment.id, cancel)
                        }
                        editMode={commentEditMode}
                        test-data="commentComponent"
                        editHistoryMode={editHistoryMode}
                        commentHistoryId={commentHistoryId}
                        toggleEditHistory={(cancel = false) =>
                          this.toggleEditHistory(comment.id, cancel)
                        }
                      />
                    );
                  })
                : ''}
            </div>
          </div>
        </Fragment>
      );
    }
    return (
      <div>
        <Spinner />
      </div>
    );
  }
}
const mapStateToprops = state => ({
  article: state.Article,
  comments: state.comments.comments,
  comment: state.comments.comment,
});

export default connect(
  mapStateToprops,
  { getOneArticle, addComment, getComments, deleteComment, editComment },
)(ReadArticle);
