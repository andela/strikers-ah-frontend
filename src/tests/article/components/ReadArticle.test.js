/* eslint-disable no-dupe-keys */
import React from 'react';
import { shallow } from 'enzyme';
import '../../../enzymeConfig';

import { ReadArticle } from '../../../components/article/ReadArticle';

const props = {
  match: {
    params: jest.fn(),
  },
  getOneArticle: jest.fn(),
  getComments: jest.fn(),
  article: {
    article: {
      taglist: ['tag1', 'tag2'],
    },
  },
};

const wrapper = shallow(<ReadArticle {...props} />);

describe('test for `ReadArticle` component', () => {
  test('should render the `ReadArticle` component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
