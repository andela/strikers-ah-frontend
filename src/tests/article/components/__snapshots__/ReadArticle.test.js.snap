// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for \`ReadArticle\` component should render the \`ReadArticle\` component 1`] = `ShallowWrapper {}`;
